api-2.2/src/main/java/org/openmrs/module/coreapps/htmlformentry/EncounterDiagnosesElement.java

```diff
@@ -61,6 +61,8 @@
     private UiUtils uiUtils;
     private String selectedDiagnosesTarget;
 
+    private String diagnosisSets;
```

Comment:
I'd recommend naming this something like 'diagnosisSetUuids" to make it clearer that this is still just a string of uuid and not a Collection<Concept>.  (Ideally this *would* be a Collection<Concept> but I realized that wouldn't make a ton of sense in this case, since you have to convert it right back to a string afterwards.

Reply:
Thank you @mogoodrich. Yes, this makes sense and perhaps, I'd simply do `diagnosisSetIds` to remain consistent with the way attributes in HFE tags are named.

Reply:
@mogoodrich, it would be _very_ nice is concept mappings could be used to specify the sets to work with. I find UUIDs very impractical and difficult to read in HFE forms.

Would that be ok with you?

The tag handler would parse the list and make sure that only UUIDs are sent over to the underlying web service (`DiagnosesFragmentController`).

Reply:
oh, absolutely @mks-d  that would be the ideal... then this *could* become a List<Concept>... the attribute on the encounterDiagnosis tag could be diagnosisSets, and you would use this existing method to parse it, as it supports our multiple methods of referencing concepts (including using mappings):

https://github.com/openmrs/openmrs-module-htmlformentry/blob/master/api/src/main/java/org/openmrs/module/htmlformentry/HtmlFormEntryUtil.java#L534

Reply:
@mogoodrich the above link is pointing in between two methods, but I'm sure @Ruhanga will figure it out ;-)

Reply:
Yes, thank you @mogoodrich.

Reply:
I have just addressed this is the commits https://github.com/openmrs/openmrs-module-coreapps/pull/199/commits/d979720b4b7b6d2b732728289d6b6c12c32ffe92 and https://github.com/openmrs/openmrs-module-coreapps/pull/199/commits/ca271c8eb14beff9177e00397f84aca2d747f6a0 .

Reply:
Sorry about the CI failure. It is a log length issue that prematurely terminates the compilation.